Tests/AK/TestTime.cpp

```diff
@@ -939,3 +939,44 @@ TEST_CASE(from_f64_seconds)
 
     EXPECT_DEATH("Converting float NaN seconds", (void)Duration::from_seconds_f64(NAN));
 }
+
+TEST_CASE(time_units)
```

Comment:
Maybe add these tests in the same commits as the implementation, so if we ever need to revert anything all relevant things are grouped together. Also a bit easier to review if you're going commit by commit :)